beanie/odm/utils/collection.py

```diff
@@ -65,7 +65,13 @@ async def collection_factory(
     # Indexed field wrapped with Indexed()
     found_indexes = [
         IndexModel(
-            [(fname, fvalue.type_._indexed[0])], **fvalue.type_._indexed[1]
+            [
+                (
+                    fvalue.alias if hasattr(fvalue, "alias") else fname,
```

Comment:
I think, here you can always use `fvalue.alias` . Pydantic already picks the right one there. Look:
https://github.com/samuelcolvin/pydantic/blob/master/pydantic/fields.py#L350

Reply:
Can we assume that fvalue is always a Field instance?

Reply:
I think yes, as we iterate over `__fields__` here which is a dictionary with `ModelField` values only:
https://github.com/samuelcolvin/pydantic/blob/master/pydantic/main.py#L276

Reply:
Done!